rtl/eclrtl/rtldynfield.cpp

```diff
@@ -1602,6 +1605,24 @@ class GeneralRecordTranslator : public CInterfaceOf<IDynamicTransform>
                 if ((field->flags & RFTMispayloadfield) == 0)
                     matchFlags |= match_keychange;
                 defaulted++;
+                // If dest field is in a nested record, we need to check that there's no "non-record" field in source matching current nested record name
+                if (destHasNested && name)
```

Comment:
minor: I don't think name can ever be null

Reply:
Probably not in the expanded case. But is there any harm in leaving the check in?

Reply:
only that it was slightly confusing, and coverity might start being picky.
Also, should it target 7.8.x?

Reply:
@richardkchapman coverity may complain that queryOriginalField(name) will crash if name is null.  Should also protect that or remove this test.

Reply:
I have moved the test. I'm wary of removing it...